File: common/src/main/java/org/bread_experts_group/breadlib/extensions/IMouseItem.java
package org.bread_experts_group.breadlib.extensions;

import net.minecraft.client.multiplayer.ClientLevel;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;

public interface IMouseItem {
	default boolean onMouseScroll(ItemStack heldStack, ClientLevel level, Player player) {
		return false;
	}

	default boolean onMouseInputPre(ItemStack heldStack, ClientLevel level, Player player) {
		return false;
	}

	default void onMouseInputPost(ItemStack heldStack, ClientLevel level, Player player) {
	}
}

File: common/src/main/kotlin/org/bread_experts_group/breadlib/extensions/item/IMouseItem.kt
package org.bread_experts_group.breadlib.extensions.item

import net.minecraft.client.multiplayer.ClientLevel
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack

interface IMouseItem {
	fun onMouseScroll(heldStack: ItemStack, level: ClientLevel, player: Player): Boolean {
		return false
	}

	fun onMouseInputPre(heldStack: ItemStack, level: ClientLevel, player: Player): Boolean {
		return false
	}

	fun onMouseInputPost(heldStack: ItemStack, level: ClientLevel, player: Player) {
	}
}
